h2/utilities.py

```diff
@@ -329,6 +329,16 @@ def _reject_pseudo_header_fields(headers, hdr_validation_flags):
             seen_pseudo_header_fields
         )
 
+    # If ':status' pseudo-header is not there in a response header, reject it
+    seen_status_field = (
```

Comment:
I wonder if it’s worth moving the evaluation of `seen_status_field` into a conditional block that only runs if this is a response header, i.e.,

```python
if hdr_validation_flags.is_response_header:
    seen_status_header = (...)
    if not seen_status_field:
        raise ProtocolError(...)
```

which saves us two unnecessary set lookups on request headers.